verification/trustpolicy/trustpolicy.go

```diff
@@ -0,0 +1,486 @@
+// Notary V2 spec rules for trusted identities
+func validateTrustedIdentities(statement TrustPolicy) error {
+
+	// If there is a wildcard in trusted identies, there shouldn't be any other
+	//identities
+	if len(statement.TrustedIdentities) > 1 && slice.Contains(slice.Wildcard, statement.TrustedIdentities) {
+		return fmt.Errorf("trust policy statement %q uses a wildcard trusted identity '*', a wildcard identity cannot be used in conjunction with other values", statement.Name)
+	}
+
+	var parsedDNs []parsedDN
+	// If there are trusted identities, verify they are valid
+	for _, identity := range statement.TrustedIdentities {
+		if identity == "" {
+			return fmt.Errorf("trust policy statement %q has an empty trusted identity", statement.Name)
+		}
+
+		if identity != slice.Wildcard {
+			i := strings.Index(identity, ":")
+			if i < 0 {
+				return fmt.Errorf("trust policy statement %q has trusted identity %q without an identity prefix", statement.Name, identity)
+			}
+
+			identityPrefix := identity[:i]
+			identityValue := identity[i+1:]
+
+			// notation natively supports x509.subject identities only
+			if identityPrefix == slice.X509Subject {
+				dn, err := pkix.ParseDistinguishedName(identityValue)
+				if err != nil {
+					return err
+				}
+				parsedDNs = append(parsedDNs, parsedDN{RawString: identity, ParsedMap: dn})
+			}
+		}
+	}
+
+	// Verify there are no overlapping DNs
+	if err := validateOverlappingDNs(statement.Name, parsedDNs); err != nil {
+		return err
+	}
+
+	// No error
+	return nil
+}
+
+// validateRegistryScopes validates if the policy document is following the
+// Notary V2 spec rules for registry scopes
+func validateRegistryScopes(policyDoc *Document) error {
+	registryScopeCount := make(map[string]int)
+
+	for _, statement := range policyDoc.TrustPolicies {
+		// Verify registry scopes are valid
+		if len(statement.RegistryScopes) == 0 {
+			return fmt.Errorf("trust policy statement %q has zero registry scopes, it must specify registry scopes with at least one value", statement.Name)
+		}
+		if len(statement.RegistryScopes) > 1 && slice.Contains(slice.Wildcard, statement.RegistryScopes) {
+			return fmt.Errorf("trust policy statement %q uses wildcard registry scope '*', a wildcard scope cannot be used in conjunction with other scope values", statement.Name)
+		}
+		for _, scope := range statement.RegistryScopes {
+			if scope != slice.Wildcard {
+				if err := validateRegistryScopeFormat(scope); err != nil {
+					return err
+				}
+			}
+			registryScopeCount[scope]++
+		}
+	}
+
+	// Verify one policy statement per registry scope
+	for key := range registryScopeCount {
+		if registryScopeCount[key] > 1 {
+			return fmt.Errorf("registry scope %q is present in multiple trust policy statements, one registry scope value can only be associated with one statement", key)
+		}
+	}
+
+	// No error
+	return nil
+}
+
+func validateOverlappingDNs(policyName string, parsedDNs []parsedDN) error {
```

Comment:
Should we rename and move this method to pkix? perhaps  `HasOverlappingDNs` ?

Reply:
This is only used in trustpolicy package, so instead of moving it into internal, we can keep it here unexported.